Assets/BossRoom/Scripts/Server/Game/Action/TrampleAction.cs

```diff
@@ -156,23 +154,44 @@ private void CollideWithVictim(ServerCharacter victim)
             victimMovement.StartKnockback(m_Parent.transform.position, Description.KnockbackSpeed, Description.KnockbackDuration);
         }
 
+        // called by owning class when parent's Collider collides with stuff
         public override void OnCollisionEnter(Collision collision)
         {
-            var actionStage = GetCurrentStage();
             // we only detect other possible victims when we start charging
-            if (actionStage != ActionStage.Charging)
+            if (GetCurrentStage() != ActionStage.Charging)
                 return;
 
-            if (m_CollidedAlready.Contains(collision.collider))
+            Collide(collision.collider);
+        }
+
+        // here we handle colliding with anything (whether a victim or not)
+        private void Collide(Collider collider)
+        {
+            if (m_CollidedAlready.Contains(collider))
                 return; // already hit them!
 
-            m_CollidedAlready.Add(collision.collider);
+            m_CollidedAlready.Add(collider);
 
-            var victim = collision.collider.gameObject.GetComponent<ServerCharacter>();
+            var victim = collider.gameObject.GetComponent<ServerCharacter>();
             if (victim)
             {
                 CollideWithVictim(victim);
             }
+            else if (!m_WasStunned)
+            {
+                // they aren't a living, breathing victim, but they might still be destructible...
+                var damageable = collider.gameObject.GetComponent<IDamageable>();
+                if (damageable != null)
+                {
+                    damageable.ReceiveHP(this.m_Parent, -Description.SplashDamage);
+
+                    // lastly, a special case: if the boss runs into certain breakables, he is stunned!
```

Comment:
Forgot to add this to my initial review, but let's also remove any mentions to the "Boss" since in theory this action could be added to any character.

Reply:
ok